src/components/fx/hover.js

```diff
@@ -237,6 +237,10 @@ function _hover(gd, evt, subplot, noHoverEvent) {
         vLinePoint: null
     };
 
+    // does subplot have one (or more) horizontal traces,
+    // (to determine whether we rotate the labels or not)
+    var hasOneHorizontalTrace = false;
```

Comment:
I'm having some seconds thoughts about this here.

Under `layout.hovermode: 'closest'`, subplots that have one (or more) "horizontal" traces get rotated labels. I'm starting to think this might be an overkill for subplots with horizontal bar traces. Perhaps, only horinzontal box and violin traces should lead to rotate the labels?

Any objections?

Reply:
I think you're right, a single trace of horizontal bars (or histogram) still looks better with unrotated labels. But as soon as you start showing multiple labels (which happens for a single trace of box or violin) it's important to rotate the labels.

Reply:
Here's what I came up with in: https://github.com/plotly/plotly.js/pull/3043/commits/d9f3db7b44c3a7ff13ee4cd79c5a99ba87a3e2fe